File: algorithms.js
//CHALLENGE1
function triangular_triplet(arr) {
    const len = arr.length;
    if (len < 3) return 0;
    arr.sort((a, b) => a - b);
    let result;
    for (let i = 0; i < len - 2; i++) {
      if (arr[i] + arr[i + 1] > arr[i + 2]) {
        result = 1;
        break;
      }
      result = 0;
    }
    return result;
  }

//CHALLENGE 2
const nested = (s) => {
    store = []
   s.split("").map(char => {
       if(['[', '{', '('].includes(char) ) {
        store.push(char)
       }
       else if([']', '}', ')'].includes(char)){
           if (store.length == 0) return 0
           x = store.pop()
           if (char== ']' && x!= '['){
                return 0;
           }else if (char == '}' && x!= '{'){
               return 0;
           }
           else if(char == ')' && x!= '('){
               return 0;
           }    
       }
    return 1      
   });  
}

//CHALLENGE 3
const findDominator = (arr) => {
    const arrTransform = arr.reduce((acc, val, index) => {
        if(acc[val]) acc[val].push(index);
        else acc[val] = [index];
        return acc
    }, {});
    let dominatorCollection;
    const isDominant = Object.values(arrTransform).some((value) => {
        dominatorCollection = value;
        return value.length > arr.length /2;
    });
    return isDominant? dominatorCollection : -1;
}

console.log(triangular_triplet([10, 2, 5, 1, 8, 20]))
console.log(nested("{[()()]}"))
console.log(findDominator([3, 4, 2, 2, -1, 3]))

File: algorithms.py
# CHALLENGE1
def triangular_triplet(arr):
    array_length = len(arr)
    if array_length<3 :
        return 0
    arr.sort()


# CHALLENGE 2
def nested(s):
    store = []
    for i in s:
        if i in ['[', '{', '(']:
            store.append(i)
        elif i in [']', '}', ')']:
            if len(store)== 0:
                return 0
            y = store.pop()
            if i == ']' and y!= '[':
                return 0
            elif i == '}' and y!= '{':
                return 0
            elif i == ')' and y!= '(':
                return 0
    return 1


# CHALLENG3
def dominator(arr):
    obj = {}
    temp = []
    for i in arr:
        occurrences = arr.count(i)
        obj[i] = occurrences
    max_key = max(obj, key = obj.get)
    max_val = max(obj.values())
    if max_val > len(arr)/2 :
        for i in range(len(arr)):
            if arr[i] == max_key:              
                temp.append(i)
        return temp
    return -1

  
print(dominator([3,4, 3, 2, 3, -1, 3, 3]))
print(nested("{[()()]}"))
print(nested("([)()]"))